src/writer.jl

```diff
@@ -0,0 +1,309 @@
+
+"""
+    write(path::AbstractString, handle::Shapefile.Handle)
+    write(path::AbstractString, geoms)
+
+Write geometries to file. `geoms` can be a table or any iterable of GeoInterface.jl
+combatible geometry objects, with `missing` values allowed.
+
+Note: As DBFTables.jl does not yet write, we can't write Table or FeatureCollection
+data besides geometries. Only .shp and .shx files are written currently.
+"""
+write(path::AbstractString, h::Handle) = write(path, h.shapes)
+function write(path::AbstractString, obj; force=false)
+    paths = _shape_paths(path)
+
+    # File existance check
+    if isfile(paths.shp)
+        if force
+            rm(paths.shp)
+        else
+            throw(ArgumentError("File already exists at `$(paths.shp)`. Use `force=true` to write anyway."))
+        end
+    end
+
+    # Handle tabular data
+    if Tables.istable(obj)
+        geomcol = GI.geometrycolumn(obj)
+        geoms = Tables.getcolumn(obj, geomcol)
+        @warn "DBFTables.jl does not yet `write`, so no .dbx file can be written"
+        # DBFTables.Table(obj) # TODO remove geom column
+        # DBFTables.write # TODO DBF write function
+    else
+        geoms = obj
+    end
+
+    # Initialisation
+    shx_indices = IndexRecord[]
+    bytes = 0
+    content_size = 0
+    first_geom = true
+    local mbr, zrange, mrange
+
+    # Open the .shp file as an IO stream
+    io = open(paths.shp, "a+")
+
+    # Write an emtpy header 
+    # We go back later and write it properly later, so we don't have to precalculate everything.
+    dummy_header = Header(;
+        filesize=0, 
+        shapecode=SHAPECODE[Missing],
+        mbr=Rect(0.0, 0.0, 0.0, 0.0),
+        zrange=Interval(0.0, 0.0),
+        mrange=Interval(0.0, 0.0),
+    )
+    bytes += Base.write(io, dummy_header)
+    header_bytes = bytes
+
+    # Detect the shape type code from the first available geometry
+    # There can only be one shape type in the file.
+    # If all values are missing, the shape type is Missing
+    if iterate(skipmissing(geoms)) == nothing
+        trait = Missing
+        hasz = hasm = false
+        shapecode = 0 
+        mbr = Rect(0.0, 0.0, 0.0, 0.0)
+        zrange = Interval(0.0, 0.0)
+        mrange = Interval(0.0, 0.0)
```

Comment:
Duplicate of empty header? Maybe not required?

Reply:
Yeah we still just writ the header again because the missing values are entered and take a tiny bit of space we have to track. But we don't get the `zrange` etc from the loop. I'll rewrite it to be clearer

Reply:
And we don't actually want to assign zeros before the loop or they will be included in the union later.

Reply:
Ok I've reorganised